### PR TITLE
fix safari rename button positioning #11348

### LINKED ISSUE #11348: [Safari] The rename button is located slightly above the normal case
### Steps to reproduce
1. Place the mouse over a file or folder

### Actual behaviour
 The rename button is located slightly above the normal case
screen shot 2014-09-29 at 09 45 21

### Server configuration
{"installed":"true","version":"7.0.2.2","versionstring":"7.0.2","edition":""}

### Client configuration
browser:Safari

Operating system: Mac 10.9.3

### PR DESCRIPTION
**Pull request by Sugaroverdose**
This fixes #11348 
Compared on safari 8, IE11, Chrome 39 - positioning looks the same

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Works fine in Chrome, but the icon isn't visible anymore in Firefox.

**Comment by Sugaroverdose** [comment]
sorry, not 'edit button', i meant 'rename button'

Checked Chrome 39,FireFox 34,Safari 8 - OS X and IE11 on WS2012R2, button is on place in all of them

**Comment by MorrisJobke** [comment]
Looks somehow broken in IE8:

ie8-rename-button

**Comment by MorrisJobke** [comment]
@Sugaroverdose Sorry for this trouble. Maybe @jancborchardt can help with the IE8 stuff.

**Comment by Sugaroverdose** [comment]
@MorrisJobke sadly, my OC8 just doesn't loads filelist on IE8, complaining about OCA.Search js problems.
But OC7 with same changes in files.css IE8 on XP displays button on same position as other browsers what i tested it on

**Comment by MorrisJobke** [comment]
I fixed them in https://github.com/owncloud/core/pull/13299 ;)

**Comment by MorrisJobke** [comment]
The errors in IE8 ;)

**Comment by DeepDiver1975** [comment]
@jancborchardt can you please help out here to get this working properly? THX

**Comment by Sugaroverdose** [comment]
@MorrisJobke looks like, it's not css bug, in OC7 dom rename button looks like this(and rendered correctly):
```
    <A class="action action-rename" href="#" original-title jQuery11000050161501470574854="240" data-action="Rename">
        <IMG class=svg src="/core/img/actions/rename.png"><SPAN> </SPAN>
    </A>
```
In OC8 dom rename button looks like this:
```
<IMG class="svg action action-rename" alt="" src="/core/img/actions/rename.png" data-action="Rename" jQuery11020452580584383972="180">
```
Interesting part: if you select element in dom explorer and push "element source with styles", you will receive correct code, not that, what you see in dom

**Comment by Sugaroverdose** [comment]
Now IE8 also positions button on it's place

File: apps/files/js/fileactions.js [hunk 1]
```diff
@@ -248,6 +248,8 @@
 			}
 			if (actionSpec.displayName) {
 				html += '<span> ' + actionSpec.displayName + '</span></a>';
+			} else {
+				html += '</a>';
```
Comment (MorrisJobke) [inline apps/files/js/fileactions.js]: simply move this out of the `else` branch and drop the `</a>` from the string in the `if` branch ;)

**Comment by MorrisJobke** [comment]
@Sugaroverdose Many thanks for spotting this! :)

cc @PVince81

**Comment by Sugaroverdose** [comment]
@MorrisJobke troubleshooting is my second name :D

**Comment by MorrisJobke** [comment]
Awesome :+1:

@Sugaroverdose Can I ask you to squash your commits into one? That would be nice. Also comment about the license of this change. MIT would fit or you sign the contributors aggreement and send it to Frank https://owncloud.org/contribute/agreement/

If you need help with anything feel free to ping me in IRC #owncloud-dev on freenode - I'm MorrisJobke there too ;)

**Comment by MorrisJobke** [comment]
@jancborchardt Review please ;)

**Comment by Sugaroverdose** [comment]
@MorrisJobke i don't know how to do that, but i'll try :)
i don't care about license, so if MIT is simplest way - it's ok)

thanks)

**Comment by MorrisJobke** [comment]
@Sugaroverdose I can also squash them on merge ;)

**Comment by Sugaroverdose** [comment]
@MorrisJobke so, i've squashed commits in separate branch and fixed 'whitespace issue', should i make a new pr?)

**Comment by MorrisJobke** [comment]
@Sugaroverdose You can force push to this branch and then github will update this PR ;)

**Comment by Sugaroverdose** [comment]
@MorrisJobke such a big new world, done)

**Comment by MorrisJobke** [comment]
@Sugaroverdose You're awesome! Nice :)

**Comment by PVince81** [comment]
@owncloud-bot ok to test

**Comment by MorrisJobke** [comment]
@PVince81 @jancborchardt Review?

**Comment by jancborchardt** [comment]
@Sugaroverdose looks ok in Firefox and Chrome still – except one thing: The clickable area doesn’t extend below the rename icon as it did before. That is, if you hover below the rename icon it doesn’t highlight. Can you fix that?

**Comment by Sugaroverdose** [comment]
@jancborchardt it can be done by increasing padding-bottom to 35px, here how it will looks like from dom perspective:
2015-01-14 4 40 51 pm

if it's ok, i'll add this to pr

**Comment by Sugaroverdose** [comment]
missed button, sry :(

**Comment by jancborchardt** [comment]
Just make sure there’s the same padding as before. :) It should go the full height of the row but no further. From your screenshot it looks like it’s bleeding into the row above.

**Comment by Sugaroverdose** [comment]
@jancborchardt that's why it's a tricky way: in this pr button already overlaps row above(floating changes element behavior to same as 'display:block;' as i remember it right), but because of 'overflow: hidden' it doesn't matter :)

**Comment by Sugaroverdose** [comment]
Tried padding-bottom:35px:
Safari 8, Chrome 39, FireFox 34 on OSX10.10 - working as expected
IE8 on WXP - hover on A element of button doesn't do anything, only IMG element clickable. but layout didn't changed.
IE11 on w2012r2 - working as expected.

**Comment by MorrisJobke** [comment]
@Sugaroverdose You can specify special rules for IE8 when you prepend the class `.ie8` in your CSS rule (the `<html>` element has this class - also for ie9 ;))

```
.ie8 .my-fancy-class {
    margin: 8px; /* a margin just for IE8 */
}
```

**Comment by Sugaroverdose** [comment]
@MorrisJobke any comments about latest commit?

**Comment by Sugaroverdose** [comment]
I've found root of issue, it's 'display: inline-block;' in table td.filename .nametext .innernametext, and looks like it's only one function - is markup destruction :) 
huge time spended on such stupid thing -_-

**Comment by MorrisJobke** [comment]
@Sugaroverdose Is this ready for review or do you want to fix something first? Maybe also squash then ;)

**Comment by MorrisJobke** [comment]
Works fine :+1: (tested with IE8, 9, FF and Chromium)

**Comment by MorrisJobke** [comment]
@owncloud-bot retest this please

**Comment by MorrisJobke** [comment]
@jancborchardt And it still has enough padding:
rename-rework

**Comment by Sugaroverdose** [comment]
@MorrisJobke it can be lowed on 1px in height to fit container size, but now it's original paddings, so i've did't touched them) 
Now this pr - is more fix than workaround: it doesn't changes much :)

**Comment by MorrisJobke** [comment]
@Sugaroverdose Yes looks good for IE8. That's how it looks in IE8 - we won't invest more time in this. It works and it is fully usable with some minor UI issues.

**Comment by jancborchardt** [comment]
@Sugaroverdose can you first squash your commits into one? :) Thanks

**Comment by MorrisJobke** [comment]
@owncloud-bot retest this please

**Comment by MorrisJobke** [comment]
@jancborchardt your turn ;)

**Comment by jancborchardt** [comment]
Will test again now. @lmaestro @georgehrke can you check as well please? This fixes https://github.com/owncloud/core/issues/11348

**Comment by georgehrke** [comment]
:+1: Issue still reproducible on latest master, fixed by this pr, renaming still works

**Comment by jancborchardt** [comment]
It unfortunately breaks though with longer filenames. These overlap with the size and modification date. This seems to be caused by the removal of the inline-block.

@Sugaroverdose can you look into it again? Sorry for the issues :)

**Comment by Sugaroverdose** [comment]
@jancborchardt So, it had function :)

**Comment by Sugaroverdose** [comment]
if this pr will be held for a week, it will contain only one char which will make everyone happy..)

**Comment by MorrisJobke** [comment]
> if this pr will be held for a week, it will contain only one char which will make everyone happy..)

Hehe :)

**Comment by Sugaroverdose** [comment]
@MorrisJobke now it's more about IE8 regressions, than safari :)

**Comment by Sugaroverdose** [comment]
@jancborchardt @MorrisJobke any comments?

**Comment by MorrisJobke** [comment]
Awesome! :) It works in IE8, IE9, Chrome and Firefox here.

**Comment by MorrisJobke** [comment]
:+1:

**Comment by LukasReschke** [comment]
Works :+1:

**Comment by MorrisJobke** [comment]
JS unit tests run just fine:

```
Coverage preprocessor:  disabled
Apps to test:  [ 'core',
  'files',
  'files_trashbin',
  { name: 'files_sharing',
    srcFiles: 
     [ 'apps/files_sharing/js/app.js',
       'apps/files_sharing/js/sharedfilelist.js',
       'apps/files_sharing/js/share.js' ],
    testFiles: [ 'apps/files_sharing/tests/js/*.js' ] },
  { name: 'files_external',
    srcFiles: 
     [ 'apps/files_external/js/app.js',
       'apps/files_external/js/mountsfilelist.js' ],
    testFiles: [ 'apps/files_external/tests/js/*.js' ] },
  { name: 'settings',
    srcFiles: [ 'settings/js/users/deleteHandler.js' ],
    testFiles: [ 'settings/tests/js/users/deleteHandlerSpec.js' ] } ]
INFO [karma]: Karma v0.12.31 server started at http://localhost:9876/
INFO [launcher]: Starting browser PhantomJS
INFO [PhantomJS 1.9.8 (Linux)]: Connected on socket ePXofAGtC015QwdNx5FE with id 9032674
WARN [web-server]: 404: /context.html//search/templates/part.results.html
.............................................................................
WARN: 'Missing plural form in language file'
.
WARN: 'Missing plural form in language file'
.
WARN: 'Missing plural form in language file'
.
WARN: 'Missing plural form in language file'
.
WARN: 'Missing plural form in language file'
.
WARN: 'Missing plural form in language file'
.
WARN: 'Missing plural form in language file'
.
WARN: 'Missing plural form in language file'
....
WARN: 'Missing plural form in language file'
.......WARN [web-server]: 404: /base/core/css/images/ui-bg_highlight-soft_100_eeeeee_1x100.png
.........................................................................
................................................................................
................................................................................
............................................
PhantomJS 1.9.8 (Linux): Executed 372 of 374 (skipped 2) SUCCESS (6.772 secs / -23693562 mins 18.861 secs)
```

**Comment by MorrisJobke** [comment]
@Sugaroverdose Many thanks for this fix! :) Sorry for the IE8 trouble :D

**Comment by Sugaroverdose** [comment]
@MorrisJobke :+1: there's one more issue #12508 which the same as #11348

**Pull request merged**